### ISSUE TITLE
vw taking invalid arguments of power on learning rate decay and learning rate

### ISSUE THREAD

**Issue by mritunjaysharma394** [VowpalWabbit/vowpal_wabbit#2340]
#### Describe the bug
The commands `vw house_dataset.txt -l 8.9` and `vw house_dataset.txt --power_t 1.5
` should produce invalid argument error since both learning rate and power on learning rate decay should be in the range [0,1]. However, the commands are working with these values
#### To Reproduce
1. `vw house_dataset.txt -l 15`
2. `vw house_dataset.txt -l -6`
3. `vw house_dataset.txt --power_t 13`
4. `vw house_dataset.txt --power_t -3`

Using any of the above commands can reproduce this behavior.
#### Expected behavior
1. Throw an error like this when learning rate entered is -4. : 
```
vw house_dataset.txt --power_t 13

finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 0
the value of power_on_learning_rate_decay must be in [0,1]
vw (parse_args.cc:1403): the value of power_on_learning_rate_decay must be in [0,1]

```
2. Or the other thing can be done is prompting and resetting it to a default value of learning rate like 0.5.
```
 vw house_dataset.txt -l 3.4
the value of learning_rate must be in [0,1]. resetting from: 3.4 to 0.5
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = house_dataset.txt
can't open 'house_dataset.txt', sailing on!
num sources = 0
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features

finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 0
```
#### Observed Behavior
1. Running flawlessly with:
```
vw house_dataset.txt --power_t 13
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 13
using no cache
Reading datafile = house_dataset.txt
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0   0.0000   0.0000        5
0.666667 1.000000            2            3.0   1.0000   0.0000        5

finished run
number of examples = 3
weighted example sum = 4.000000
weighted label sum = 2.000000
average loss = 0.750000
best constant = 0.500000
best constant's loss = 0.250000
total feature number = 15

```

#### Environment
VW 8.8.0 on Linux with the Python bindings

#### Status 
I have solved this bug and will be creating a PR for the same

**Comment by mritunjaysharma394** [comment]
This PR #2341 is trying to resolve the above issue

**Comment by lokitoth** [comment]
Hi @mritunjaysharma394,

Thank you for the report. 

I am wondering, how did you choose the interval [0, 1] rather than, rather than (1, +inf) for learning rate, and [0, +inf) for power_t?

For example, if you set learning_rate to zero, you will never converge, because the magnitude of the update will always be zero.

**Comment by mritunjaysharma394** [comment]
Hi @lokitoth,

As far as power_t is concerned, while I was going to the tutorial on https://vowpalwabbit.org/tutorials/getting_started.html , I came across this - 

![Screenshot from 2020-03-16 22-06-50](https://user-images.githubusercontent.com/45476902/76780270-bd016600-67d2-11ea-9c4d-b12e12249480.png)

Here it was mentioned that the value of power_t can vary from [0,1], and when I tried to run the commands with a negative value or a value greater than 1 of power_t, it ran flawlessly. 

Also in the tutorial, it was mentioned that default learning_rate is 0.5.
so I regret the error I made there and the learning_rate can be in the range (0,1) instead of [0,1].
I am not sure but I came across these when I was going through the tutorial and so thought of reporting it. Would like to know more if I am interpreting it wrong? 

![Screenshot from 2020-03-16 22-06-26](https://user-images.githubusercontent.com/45476902/76780840-b2939c00-67d3-11ea-9083-be0eb7b60559.png)

**Comment by jackgerrits** [comment]
@lokitoth Can you confirm the values in the tutorial are incorrect?

**Comment by jackgerrits** [comment]
I spoke with Marco and power_t can take all real numbers. I have updated the tutorial to reflect this (pending deploy)

**Comment by mritunjaysharma394** [comment]
Thank you so much for pointing it out. I am closing the issue.
